### PR TITLE
Change Stucture.output behavior/return type

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

### Changed
- `Structure.output`'s type is now `BaseArtifact` and raises an exception if the output is `None`.
Cuts out `None` checks post Structure run
## Issue ticket number and link
NA

### CODE REVIEW

File: griptape/structures/pipeline.py [hunk 1]
```diff
@@ -53,8 +53,8 @@ def insert_task(self, parent_task: BaseTask, task: BaseTask) -> BaseTask:
     def try_run(self, *args) -> Pipeline:
         self.__run_from_task(self.input_task)
 
-        if self.conversation_memory and self.output is not None:
-            run = Run(input=self.input_task.input, output=self.output)
+        if self.conversation_memory and self.output_task.output is not None:
```
Comment (cjkindel) [inline griptape/structures/pipeline.py:56]: Is there any desire to consolidate the conversation memory logic of all the structures (L56-59 here, L117-120 in workflow)? Noticed the same code when tinkering with a custom Agent locally and now this PR is surfacing it again.

Comment (collindutter) [reply griptape/structures/pipeline.py:56]: Sure thing, it would consolidate nicely into `Structure.after_run`. Would you mind making that PR?

Comment (collindutter) [reply griptape/structures/pipeline.py:56]: Actually since there's no change in behavior I can just cram it into this one.

**Review by cjkindel — APPROVED** [review]

File: griptape/structures/structure.py [hunk 2]
```diff
@@ -163,6 +165,11 @@ def before_run(self, args: Any) -> None:
 
     @observable
     def after_run(self) -> None:
+        if self.conversation_memory and self.output_task.output is not None:
```
Comment (vachillo) [inline griptape/structures/structure.py:168]: do you need the second `is not None` check now?

Comment (collindutter) [reply griptape/structures/structure.py:168]: There were some tests that relied on `output_task.output` being `None` which is still a valid state. Didn't feel like refactoring those.

**Pull request merged**